proto/cosmos/upgrade/v1beta1/upgrade.proto

```diff
@@ -40,6 +40,39 @@ message Plan {
   // If this field is not empty, an error will be thrown.
   google.protobuf.Any upgraded_client_state = 5
       [deprecated = true];
+
+  // Optional: Upgrade contains additional instructions for the devops or a hypervisor.
+  // App specific instructions are handled by the `info` attribute. Here we provide
+  // information such as pre-upgrade or post-upgrade commands.
+  UpgradeInstructions Upgrade = 6;
```

Comment:
It might actually be desirable to extract the `platform` field all the way out to the `UpgradeInstructions` message and make the `UpgradeInstructions Upgrade` a `repeated` field with a comment that it SHOULD have only one entry per platform. This would allow for different pre/post-run commands that can be tailored for specific environments.

Also, in order to help minimize on-chain data, should we throw this in a `oneof` and allow a url to be provided in place of the full structure here? The idea is that the url should return the same structure, but really, for chains that don't use `Cosmovisor`, it could return whatever they want. If the `UpgradeInstructions` are provided directly, they'll be serialized as json and added to the `upgrade-info.json` file in the `upgrade` field. If a url is provided, that url will be added to the `upgrade-info.json` file as `upgrade_url`.

Lastly, I feel like this field would be better named `instructions`. The proto file is named `upgrade.proto` and so is the module for it. The message the field is in is `Plan` but is sometimes called an "upgrade plan." The `info` field is sometimes called "upgrade info." Basically, the word "upgrade" is already used in a few ways and having a field with that name can easily cause more confusion.

Reply:
> Also, in order to help minimize on-chain data, should we throw this in a oneof and allow a url to be provided in place of the full structure here? The idea is that the url should return the same structure, but really, for chains that don't use Cosmovisor, it could return whatever they want. If the UpgradeInstructions are provided directly, they'll be serialized as json and added to the upgrade-info.json file in the upgrade field. If a url is provided, that url will be added to the upgrade-info.json file as upgrade_url.

I was thinking about this part a bit further with extra consideration for the first upgrade using new definitions.

If we allow the a url in the `info` field to return the newly defined `UpgradeInstructions` structure (e.g. as JSON), then as long as the nodes are running updated versions of Cosmovisor, an update can be done using these new proto fields without needing the chain software to already be using the new version. Otherwise, these new fields cannot be used for the next upgrade; they can only be used for upgrades **after** the one that adds these fields to the protos.

The downside is the further overloading of the `info` field.